docs/language-usage.rst

```diff
@@ -0,0 +1,279 @@
+
+- `analyzeEntities`_
+- `analyzeSentiment`_
+- `annotateText`_
+
+and each method uses a `Document`_ for representing text. To
+create a :class:`~gcloud.language.document.Document`,
+
+  .. code-block:: python
+
+     >>> text_content = (
+     ...     'Google, headquartered in Mountain View, unveiled the '
+     ...     'new Android phone at the Consumer Electronic Show.  '
+     ...     'Sundar Pichai said in his keynote that users love '
+     ...     'their new Android phones.')
+     >>> document = client.document_from_text(text_content)
+
+By using :meth:`~gcloud.language.client.Client.document_from_text`,
+the document's type is plain text:
+
+  .. code-block:: python
+
+     >>> document.doc_type == language.Document.PLAIN_TEXT
+     True
+
+In addition, the document's language defaults to the language on
+the client
+
+  .. code-block:: python
+
+     >>> document.language
+     'en'
+     >>> document.language == client.language
+     True
+
+In addition, the
+:meth:`~gcloud.language.client.Client.document_from_html`,
+factory can be used to created an HTML document. In this
+method and the from text method, the language can be
+over-ridden:
+
+  .. code-block:: python
+
+     >>> html_content = """\
+     ... <html>
+     ...   <head>
+     ...     <title>El Tiempo de las Historias</time>
+     ...   </head>
+     ...   <body>
+     ...     <p>La vaca salt&oacute; sobre la luna.</p>
+     ...   </body>
+     ... </html>
+     ... """
+     >>> document = client.document_from_html(html_content,
+     ...                                      language='es')
+
+The ``language`` argument can be either ISO-639-1 or BCP-47 language
+codes; at the time, only English, Spanish, and Japanese `are supported`_.
+However, the ``analyzeSentiment`` method `only supports`_ English text.
+
+.. _are supported: https://cloud.google.com/natural-language/docs/
+.. _only supports: https://cloud.google.com/natural-language/reference/rest/v1beta1/documents/analyzeSentiment#body.request_body.FIELDS.document
+
+The document type (``doc_type``) value can be one of
+:attr:`Document.PLAIN_TEXT <gcloud.language.document.Document.PLAIN_TEXT>` or
+:attr:`Document.HTML <gcloud.language.document.Document.HTML>`.
+
+In addition to supplying the text / HTML content, a document can refer
+to content stored in `Google Cloud Storage`_. We can use the
+:meth:`~gcloud.language.client.Client.document_from_blob` method:
+
+  .. code-block:: python
+
+     >>> document = client.document_from_blob(bucket='my-text-bucket',
+     ...                                      blob='sentiment-me.txt')
+     >>> document.gcs_url
+     'gs://my-text-bucket/sentiment-me.txt'
+     >>> document.doc_type == language.Document.PLAIN_TEXT
+     True
+
+and the :meth:`~gcloud.language.client.Client.document_from_uri`
+method. In either case, the document type can be specified with
+the ``doc_type`` argument:
+
+  .. code-block:: python
+
+     >>> gcs_url = 'gs://my-text-bucket/sentiment-me.txt'
```

Comment:
Can we auto-detect HTML based on the content-type or file extension (`.html`, `.htm`) ? If we do that, then life gets a bit easier on these, right?